backend/src/main/scala/cromwell/backend/io/JobPaths.scala

```diff
@@ -30,12 +31,21 @@ trait JobPaths {
 
   def workflowPaths: WorkflowPaths
   def returnCodeFilename: String = "rc"
-  def stdoutFilename: String = "stdout"
-  def stderrFilename: String = "stderr"
+  def defaultStdoutFilename = "stdout"
+  def defaultStderrFilename = "stderr"
+
+  final def stdoutFilename: String = jobKey.node match {
+    case t: TaskCallNode => t.callable.stdoutRedirection.getOrElse(defaultStdoutFilename)
+    case _ => ??? // TODO: JobKey should be typesafe
```

Comment:
could `node` be `node: TaskCallNode` on CallKey?

Reply:
No it could also be a `WorkflowCallNode` (for sub workflows).
But it should be a `TaskCallNode` on `BackendJobDescriptorKey`